app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -447,6 +455,44 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
         if (toShow) pocketRepoCache.freeze() else pocketRepoCache.unfreeze()
     }
 
+    fun WebBackForwardList.toList(): List<String> {
+        return List(size) {
+            getItemAtIndex(it).originalUrl
+        }
+    }
+
+    /**
+     * JavascriptInterface for handling when to map back click to ESC vs perform back in history.
+     * Clicking remote back will remap to ESC until the active DOM element is an item in YouTube's guidelist-
+     * then it will go back in the history stack.
+     */
+    inner class YouTubeBackHandler {
+        @JavascriptInterface
+        fun youtubeBack(sendBack: Boolean) {
+            if (sendBack) {
+                Handler(Looper.getMainLooper()).post {
```

Comment:
Creating a new Handler might be expensive so we wouldn't want to do it each time this is called. Maybe `launch(UI)`?